collector/openhardwaremonitor.go

```diff
@@ -0,0 +1,82 @@
+package collector
+
+import (
+	"strconv"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector("openhardwaremonitor", NewOpenHardwareMonitorCollector)
+}
+
+// A openHardwareMonitorCollector is a Prometheus collector for WMI Sensor metrics
+type SensorCollector struct {
+	SensorType *prometheus.Desc
+	Identifier *prometheus.Desc
+	Parent     *prometheus.Desc
+	Name       *prometheus.Desc
```

Comment:
Are the sensor names dynamic? If the names are static and known before execution, it may be worth moving the `name` from the labels to the metric name.

Reply:
The sensor name is dynamic... Talking about this, I wanted to move SensorType from the label to the metric name, as it has [8 predefined types with known units](http://openhardwaremonitor.org/wordpress/wp-content/uploads/2011/04/OpenHardwareMonitor-WMI.pdf), but couldn't find a way to make the metric name change accordingly. I'm new to Go (and this codebase), so maybe it's just ignorance, but is there any way to resolve the metric name during execution?

Reply:
The `NewOpenHardwareMonitorCollector` function is called on exporter initialisation, so we can't dynamically create metric names at runtime.

As the types are static, they could be included in the metric names. Here's two as an example:

```go
	return &SensorCollector{
		VoltageValue: prometheus.NewDesc(
			prometheus.BuildFQName(Namespace, subsystem, "sensor_value_volts"),
			"Value from an OpenHardwareMonitor sensor in volts",
			[]string{"parent", "index", "name"},
			nil,
		),
	}, nil
	return &SensorCollector{
		TemperatureValue: prometheus.NewDesc(
			prometheus.BuildFQName(Namespace, subsystem, "sensor_value_degrees"),
			"Calue from an OpenHardwareMonitor sensor in degrees Celsius",
			[]string{"parent", "index", "name"},
			nil,
		),
	}, nil
```